### PR TITLE
Quote symbols whose inspect requires it

### PR DESCRIPTION
**Pull request by ryanseys**
`Symbol#inspect` and the `p`/inspect rendering of a symbol emitted `:` followed
by the raw name, dropping the quotes CRuby adds when the name is not a plain
identifier:

```ruby
p :"a b"   # => :a b    (want :"a b")
p :"1x"    # => :1x     (want :"1x")
```

A runtime helper `sp_sym_inspect` (with a name-only `sp_sym_inspect_name`)
decides quoting once: a symbol prints bare when its name is a plain identifier
or constant, an `@ivar`/`@@cvar`/`$gvar`, a name ending in `?`/`!`/`=`, or a
known operator method name; otherwise it is quoted like a string. The scalar
`p`, `Symbol#inspect`, the poly inspect arm, and the symbol-array inspect all
route through it. Hash symbol keys keep the short `name:` form and quote the
name when needed (`"k space": ...`) via `sp_sym_inspect_key`.

`sp_sym_inspect` was referenced by a codegen path but never defined; it now
exists, closing that latent gap.

`test/symbol_inspect_quoting.rb` exhaustively covers the full operator set,
every quoting category (sigils, suffixes, constants), edge cases (empty,
escaped quote, embedded newline), the `.inspect` string path, symbol arrays,
and quoted hash keys. Full suite green.

### CODE REVIEW

File: lib/sp_str.c [hunk 1]
```diff
@@ -43,6 +43,37 @@ const char*sp_str_concat4(const char*a,const char*b,const char*c,const char*d){i
...
+/* A symbol prints without quotes when its name is a plain identifier (an
+   @ivar / @@cvar / $gvar, or a bare name optionally ending in ? ! =) or a
+   known operator method name; otherwise it is quoted like a string: :"a b". */
+mrb_bool sp_sym_plain_name_p(const char *p, mrb_bool allow_suffix) {
+  if (!(isalpha((unsigned char)*p) || *p == '_')) return FALSE;
+  p++;
+  while (isalnum((unsigned char)*p) || *p == '_') p++;
+  if (allow_suffix && (*p == '?' || *p == '!' || *p == '=')) p++;
+  return *p == '\0';
+}
```
[the remark this answers is not in the document]
Comment (ryanseys) [reply lib/sp_str.c:59]: Done — switched the classification to explicit ASCII ranges (cast to `unsigned char`) so it is locale-independent regardless of `LC_CTYPE`. Behavior is unchanged today (the runtime never calls `setlocale`, so it always ran in the C locale), but this removes the latent dependency and matches the intent of identifier classification. One scope note: a multibyte-identifier symbol (e.g. `:"あ"`) stays quoted — aligning multibyte identifier rules with CRuby is a separate pre-existing gap, left out of scope here.

File: lib/sp_runtime.h [hunk 2]
```diff
@@ -2524,7 +2527,7 @@ static inline const char *sp_poly_inspect(sp_RbVal v) {
     case SP_TAG_FLT:  return sp_float_to_s(v.v.f);
     case SP_TAG_BOOL: return v.v.b ? SPL("true") : SPL("false");
     case SP_TAG_NIL:  return SPL("nil");
-    case SP_TAG_SYM:  return sp_str_concat(SPL(":"), sp_sym_to_s((sp_sym)v.v.i));
+    case SP_TAG_SYM:  return sp_sym_inspect_name(sp_sym_to_s((sp_sym)v.v.i));
```
[the remark this answers is not in the document]
Comment (ryanseys) [reply lib/sp_runtime.h]: Done — routed the `SP_TAG_SYM` arm through the existing `sp_sym_inspect((sp_sym)v.v.i)` helper.
